server/src/test/java/org/opensearch/ingest/AbstractBatchingProcessorTests.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.ingest;
+
+import org.opensearch.test.OpenSearchTestCase;
+import org.junit.Before;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Consumer;
+
+import org.mockito.ArgumentCaptor;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class AbstractBatchingProcessorTests extends OpenSearchTestCase {
+
+    private static final String DESCRIPTION = "description";
+    private static final String TAG = "tag";
+
+    @Before
+    public void setup() {}
+
+    public void testBatchExecute_emptyInput() throws Exception {
+        DummyProcessor processor = spy(new DummyProcessor(TAG, DESCRIPTION, 3));
+        Consumer<List<IngestDocumentWrapper>> handler = (results) -> { assertTrue(results.isEmpty()); };
+        processor.batchExecute(Collections.emptyList(), handler);
+        verify(processor, never()).internalBatchExecute(anyList(), any());
```

Comment:
This is an implementation detail. You should test the behavior, not the specific implementation. In this case, the behavior to test is that if given empty input then the passed in `Consumer` is invoked with an empty list.